src/west/app/project.py

```diff
@@ -640,6 +645,45 @@ def _die_if_manifest_project_filter(self, action):
                      'the manifest while projects are made inactive by the '
                      'project filter.')
 
+    def _untracked(self):
+        ppaths = []
+        untracked = []
+        for project in self._projects(None):
+            # We do not check for self.manifest.is_active(project) because
+            # inactive projects are still considered "tracked folders"
+            ppaths.append(Path(project.abspath).resolve(strict=False))
```

Comment:
I think these `ppaths` needs to be sorted so parent directories are checked first, otherwise in `_find_untracked` the `elif e in ppath.parents:` is triggered before `if ppath == e:`

```suggestion
        for project in self._projects(None):
            # We do not check for self.manifest.is_active(project) because
            # inactive projects are still considered "tracked folders"
            ppaths.append(Path(project.abspath).resolve(strict=False))

        ppaths.sort() # make sure hierarchy is sorted
```

Reply:
You're very right. Fixed.